### PR TITLE
Correct shell scripts issues, format shell scripts

### PR DESCRIPTION
**Pull request by IgorTodorovskiIBM**
Also introduces the script `tools/format_and_check_scripts.sh` which depends on shfmt (go tool) and shellcheck (only available on non-z/OS since its written in haskell)

Related to https://github.com/ZOSOpenTools/meta/pull/482

### CODE REVIEW

**Comment by IgorTodorovskiIBM** [comment]
Exploring the options for shellcheck. I had passed -o all initially which triggered it to also change all information/warning cases. I'm using this currently:
```
shellcheck --shell sh -e SC2034,SC2048  --severity=error
```

File: bin/zopen-help2man [hunk 1]
```diff
@@ -67,23 +67,20 @@ for tooldir in ${tooldirs}; do
   for tool in $tooldir/zopen*; do
     basetool=$(basename ${tool})
     help2man ${help2manopts} "${tool}" >"${mandir}/${basetool}.1"
-    if  $( "${tool}" --help >/dev/null 2>&1 ) ; then
+    if $("${tool}" --help >/dev/null 2>&1); then
       # Tool has implemented --help
-      if  $( "${tool}" --version >/dev/null 2>&1 ) ; then
+      if $("${tool}" --version >/dev/null 2>&1); then
```
Comment (MikeFultonDev) [inline bin/zopen-help2man:72]: i kinda like a space here - but if others don't, i'm ok to change... e.g. i prefer `$( <cmd> )` rather than `$(<cmd>)`

**Review by MikeFultonDev — APPROVED** [review]
changes seem reasonable. 
maybe this would be a good thing to put a test or two into metaport first to validate nothing breaks?

**Review by MikeFultonDev — APPROVED** [review]
Lgtm

**Review by MikeFultonDev — COMMENTED** [review]
Does -z test for both unset and equal to ‘’ like the old code ?

**Comment by IgorTodorovskiIBM** [comment]
> Does -z test for both unset and equal to ‘’ like the old code ?

Yep:
```
[ITODORO@ZOSCAN2B ~/projects/meta_testme]$ unset BLA; test -z "$BLA"; echo $?
0
[ITODORO@ZOSCAN2B ~/projects/meta_testme]$ BLA=""; test -z "$BLA"; echo $?
0
[ITODORO@ZOSCAN2B ~/projects/meta_testme]$ BLA=''; test -z "$BLA"; echo $?
0
[ITODORO@ZOSCAN2B ~/projects/meta_testme]$
```

**Review by MikeFultonDev — APPROVED** [review]
Lgtm

**Review by MikeFultonDev — APPROVED** [review]
Lgtm

**Pull request merged**